libfreerdp/core/nego.c

```diff
@@ -677,20 +677,16 @@ static BOOL nego_read_request_token_or_cookie(rdpNego* nego, wStream* s)
 	if (Stream_GetRemainingLength(s) < 15)
 		return TRUE;
 
-	if (!memcmp(Stream_Pointer(s), "Cookie: msts=", 13))
+	if (memcmp(Stream_Pointer(s), "Cookie: mstshash=", 17) != 0)
```

Comment:
Sorry to ask, but does this break the old behaviour or do I miss something here?

Reply:
The old behaviour was broken.
Right know if the Cookie is not prefixed by 'Cookie: mstshash=' we treat is as token.
If you see the code that sends the cookie and token (nego_send_negotiation_request) the cookie is prefixed by `Cookie: mstshash=` in case of token we skiped 13 bits but never prefixed the Cookie again. So if Token is know set it will also contains this 'msts' prefix. As well if the token will contains 'loadbalance info' like in my case then  we also will just read it to the token.